### ISSUE TITLE
getTypes should not contain nullable entries

### ISSUE THREAD

**Issue by mvriel** [phpDocumentor/phpDocumentor#2248]
While upgrading PHPStan's coverage to level 6 it reported that entries in the `getTypes` method of the ConstantDescriptor could be null values (https://github.com/phpDocumentor/phpDocumentor/pull/2239#discussion_r373789636); this should not be the case as it makes it harder to iterate that collection.

For this issue, we need to investigate why PHPStan believes there could be null values and ensure there aren't

**Comment by jaapio** [comment]
I'm not sure if null values are even possible with the new reflection component. I think it could be part of our legacy system, needs to be checked to make sure I am right

**Comment by mvriel** [comment]
Yeah, I did not expect this either. So I am really curious how this happens

**Comment by jaapio** [comment]
@orklah I think you already addressed this issue?

**Comment by orklah** [comment]
@jaapio I did not try to tackle this yet but this is a good idea for a next step :)

One of the source cause is there:
https://github.com/phpDocumentor/phpDocumentor/blob/master/src/phpDocumentor/Descriptor/ConstantDescriptor.php#L85
if somehow the instanceof is failing, we will return null, and the method getTypes on this descriptor is just a return [$this->getType()] so it will return null as well.

I don't know if it actually happens though. A couple of Assert and the CI should tell us what's going on
